src/kbmod/reprojection.py

```diff
@@ -191,35 +233,64 @@ def _reproject_work_unit(work_unit, common_wcs, frame="original"):
 
         psf = images[indices[0]].get_psf()
 
-        new_layered_image = LayeredImage(
-            science_add,
-            variance_add,
-            mask_add,
-            psf,
-            time,
-        )
-        stack.append_image(new_layered_image, force_move=True)
+        if write_output:
+            _write_images_to_shard(
+                science_add=science_add,
+                variance_add=variance_add,
+                mask_add=mask_add,
+                psf=psf,
+                wcs=common_wcs,
+                obstime=time,
+                obstime_index=obstime_index,
+                indices=indices,
+                directory=directory,
+                filename=filename,
+            )
+        else:
+            new_layered_image = LayeredImage(
+                science_add,
+                variance_add,
+                mask_add,
+                psf,
+                time,
+            )
+            stack.append_image(new_layered_image, force_move=True)
 
-    per_image_wcs = work_unit._per_image_wcs
-    per_image_ebd_wcs = work_unit._per_image_ebd_wcs
+    if write_output:
+        new_work_unit = copy(work_unit)
+        new_work_unit._per_image_indices = unique_obstime_indices
+        new_work_unit.wcs = common_wcs
+        new_work_unit.reprojected = True
 
-    new_wunit = WorkUnit(
-        im_stack=stack,
-        config=work_unit.config,
-        wcs=common_wcs,
-        constituent_images=work_unit.constituent_images,
-        per_image_wcs=per_image_wcs,
-        per_image_ebd_wcs=per_image_ebd_wcs,
-        per_image_indices=unique_obstime_indices,
-        geocentric_distances=work_unit.geocentric_distances,
-        reprojected=True,
-    )
+        hdul = new_work_unit.metadata_to_primary_header()
+        hdul.writeto(os.path.join(directory, filename))
+    else:
+        per_image_wcs = work_unit._per_image_wcs
+        per_image_ebd_wcs = work_unit._per_image_ebd_wcs
```

Comment:
I don't think you need to set these here. I think you can just reference them directly in the constructor parameters below.